turbo/transactions/tracing.go

```diff
@@ -211,10 +211,7 @@ func ExecuteTraceTx(
 
 	result, err := execCb(evm, refunds)
 	if err != nil {
-		if streaming {
-			stream.WriteArrayEnd()
```

Comment:
where these methods are called now? are they become obsolete?

Reply:
and if do WriteNil here, should not we do same on each error above?

Reply:
In case of any error occurring while the JSON stream is some inconsistent state, the stack-based json stream takes care of writing nil or closing pending objects by calling `WriteNil` or `WriteArrayEnd`/`WriteObjectEnd`.

However, you still have to call all the stream methods (included `WriteNil`/`WriteArrayEnd`/`WriteObjectEnd`) to open/close JSON arrays/objects while populating the stream: the stack-based implementation really kicks in when some error occurs.

Reply:
> and if do WriteNil here, should not we do same on each error above?

this specific `WriteNil` call is not for a streaming API, while the ones above are for streaming APIs